src/components/ha-form/compute-initial-ha-form-data.ts

```diff
@@ -34,12 +34,37 @@ export const computeInitialHaFormData = (
       };
     } else if ("selector" in field) {
       const selector: Selector = field.selector;
-      if ("boolean" in selector) {
+
+      if ("device" in selector) {
+        if (selector.device.multiple) {
+          data[field.name] = [];
+        } else {
+          data[field.name] = "";
+        }
```

Comment:
Slightly more readable
```suggestion
        data[field.name] = selector.device.multiple ? [] : "";
```